block/block.md

```diff
@@ -1 +1,105 @@
+dalc|da.DataAvailabilityLayerClient|the data availability light client used to submit and retrieve blocks to DA network
+blockstore|*goheaderstore.Store[*types.Block]|to retrieve blocks gossiped over the p2p network
+
+Block manager configuration options:
+
+|Name|Type|Description|
+|-----|-----|-----|
+|BlockTime|time.Duration|time interval used for block production and block retrieval from block store (default: 1 second)|
+|DABlockTime|time.Duration|time interval used for both block publication to DA network and block retrieval from DA network (default: 15 seconds)|
+|DAStartHeight|uint64|block retrieval from DA network starts from this height|
+|NamespaceID|bytes|8 `byte` unique identifier of the rollup|
+
+### Block Production
+
+When the full node is operating as a sequencer (aka aggregator), the block manager runs the block production logic. There are two modes of block production, which can be specified in the block manager configurations: `normal` and `lazy`.
+
+In `normal` mode, the block manager runs a timer, which is set to the `BlockTime` configuration parameter, and continuously produces blocks at `BlockTime` intervals.
+
+In `lazy` mode, the block manager starts building a block when any transaction becomes available in the mempool. After the first notification of the transaction availability, the manager will wait for a 1 second timer to finish, in order to collect as many transactions from the mempool as possible. The 1 second delay is chosen in accordance with the default block time of 1s. The block manager also notifies the full node after every lazy block building.
+
+#### Building the Block
+
+The block manager of the sequencer nodes performs the following steps to produce a block:
+
+* Call `CreateBlock` using executor
+* Sign the block using `signing key` to generate commitment
+* Call `ApplyBlock` using executor to generate an updated state
+* Save the block, validators, and updated state to local store
+* Add the newly generated block to `pendingBlocks` queue
+* Publish the newly generated block to channels to notify other components of the sequencer node (such as block and header gossip)
+
+### Block Publication to DA Network
+
+The block manager of the sequencer full nodes regularly publishes the produced blocks (that are pending in the `pendingBlocks` queue) to the DA network using the `DABlockTime` configuration parameter defined in the block manager config. In the event of failure to publish the block to the DA network, the manager will perform `maxSubmitAttempts` (default is 30) attempts and an exponential backoff interval between the attempts. The exponential backoff interval starts off at 100 milliseconds and it doubles in the next attempt and capped at `DABlockTime`. A successful publish event leads to the emptying of `pendingBlocks` queue and a failure event leads to proper error reporting and without emptying of `pendingBlocks` queue.
```

Comment:
RE: `maxSubmitAttempts` we should just reference the variable and it's function. Adding the values in the specs adds another thing to track when things are updated. It is fine to use an example, like in the block times for rollup and DA layer.

Instead of listing the values of the constants/variables in the specs, we should reference where in the code the defaults are defined so that they can be referenced easily. 

This comment applies to all variables in the spec.